### PR TITLE
[cli]: wait waitUntil promises in Serverless

### PR DESCRIPTION
**Pull request by Kikobeats**
The `waitUntil` in serverless is not waited before closing the dev server.

This wrong behavior is just happening for without `'edge'` because under `'edge'` uses `runServer` helpers from `edge-runtime` which already is taking care of promises to wait before closing the server:

- https://github.com/vercel/edge-runtime/blob/352af9028a260b6ec61dda721aa384b19f56c7b4/packages/runtime/src/server/run-server.ts#L34
- https://github.com/vercel/edge-runtime/blob/352af9028a260b6ec61dda721aa384b19f56c7b4/packages/vm/src/edge-vm.ts#L248

I fixed the problem by just tracking promises to be waited on from userspace, making it possible to wait on them before closing the server.

**TODO**

- [x] add a test
- [x] show a warning if waitUntil is taking too much time (https://github.com/vercel/vercel/blob/d568e3466e57e0a80652d9d5c09fbf94f1f28e75/packages/node/src/edge-functions/edge-handler.mts#L176)

### CODE REVIEW

File: packages/node/src/utils.ts
Comment (styfle) [inline packages/node/src/utils.ts]:
```suggestion
export const waitUntilWarning = (entrypointPath: string) =>
```

Lets follow the naming convention for functions

File: packages/node/test/dev-fixtures/edge-waituntil.js
Comment (styfle) [inline packages/node/test/dev-fixtures/edge-waituntil.js:1]: Why is this file deleted?

File: packages/node/test/dev-fixtures/serverless-fetch.js
Comment (styfle) [inline packages/node/test/dev-fixtures/serverless-fetch.js:1]: Also, why is this file deleted?

File: packages/node/test/dev-fixtures/edge-waituntil.js
Comment (Kikobeats) [reply packages/node/test/dev-fixtures/edge-waituntil.js:1]: there is a new file for testing waitUntil, also another one for Node.js runtime

File: packages/node/test/dev-fixtures/serverless-fetch.js
Comment (Kikobeats) [reply packages/node/test/dev-fixtures/serverless-fetch.js:1]: Not used

File: packages/node/test/dev-fixtures/web-handlers-node-wait-until.js
Comment (styfle) [inline packages/node/test/dev-fixtures/web-handlers-node-wait-until.js:1]: Is this file used? I'm not seeing it referenced in `packages/node/test/unit/dev.test.ts` 🤔

Comment (Kikobeats) [reply packages/node/test/dev-fixtures/web-handlers-node-wait-until.js:1]: good catch; removed

**Review by styfle — APPROVED** [review]
LGTM

You should probably get a codeowner's approval as well

File: packages/node/src/serverless-functions/helpers-web.ts
Comment (Ethan-Arrowood) [inline packages/node/src/serverless-functions/helpers-web.ts:12]: I've never seen a class defined inline like this before. Maybe we can do this outside of the function call and then pass it in like normal?

File: packages/node/src/utils.ts
Comment (Ethan-Arrowood) [inline packages/node/src/utils.ts:17]: is the path guaranteed to use `'/'`? Or should this be https://nodejs.org/api/path.html#pathsep

File: packages/node/src/serverless-functions/helpers-web.ts
Comment (Kikobeats) [reply packages/node/src/serverless-functions/helpers-web.ts:12]: it isn't part of the scope of this PR

**Pull request closed without merging**
